### PR TITLE
Features added

### PR DESCRIPTION
**Pull request by QuickLearner171998**
#### Reference Issues/PRs

#11 

#### What does this implement/fix? Explain your changes.
Following features are added:
1) Threshold can now be set by the user.
2) Pascal VOC saving format.
3) Add all classes at once without selecting each manually.
4) Option to choose between tensorflow and keras models.
5) Display accuracy and model name.

### CODE REVIEW

**Comment by virajmavani** [comment]
Hi @QuickLearner171998! Thank you for lending a hand in implementing these features. I haven't set-up CI yet so I will have to manually test it. Once done, I will be able to merge the changes.

Thank you!

File: main.py [hunk 1]
```diff
@@ -20,26 +18,26 @@
...
-
-model_path = os.path.join('.', 'snapshots', 'resnet50_coco_best_v2.1.0.h5')
-
-model = models.load_model(model_path, backbone_name='resnet50')
+# make sure the file is inside semi-auto-image-annotation-tool-master
+import pathlib
+cur_path = pathlib.Path(__file__).parent.absolute()
+# cur_path = r"semi-auto-image-annotation-tool-master"
+sys.path.append(cur_path)
+os.chdir(cur_path)
```
Comment (virajmavani) [inline main.py:30]: Have you tried it across platforms? I encountered an error here running it on Debian.

`TypeError: argument should be string, bytes or integer, not PosixPath`

Change line 27, 
From:
`cur_path = pathlib.Path(__file__).parent.absolute()`
To:
`cur_path = pathlib.Path(__file__).parent.absolute().as_posix()`

This fixes the problem.

This is running on Python3.5 which is the minimum required Python version.

File: main.py [hunk 2]
```diff
@@ -20,26 +18,26 @@
...
 
 
 class MainGUI:
     def __init__(self, master):
+
+        # to choose between keras or tensorflow models
+        self.kera = 1  # default
+        self.tensorf = 0
```
Comment (virajmavani) [inline main.py:38]: You might want to use PEP8 standards here.

`self.keras_` and `self.tensorflow_` can be good candidates for the variable names.

File: main.py [hunk 3]
```diff
@@ -175,15 +199,27 @@ def __init__(self, master):
...
+
+    def available_models(self):
+        self.models_dir = os.path.join(cur_path, 'snapshots')
+        # only for keras and tf
+        model_categ = os.listdir(self.models_dir)
+        # creating all model options list
+        model_names = []
+        for categ in model_categ:
+            for name in os.listdir(os.path.join(self.models_dir , categ)):
+                model_names.append(os.path.join(categ,name))
```
Comment (virajmavani) [inline main.py:215]: After adding the README.md file in the snapshots directory, the listdir for model_categ also returns README.md and the iterator cannot then list it further as it is a file.

Changing line 210,
From:
`model_categ = os.listdir(self.models_dir) `
To:
`model_categ = [dir_ for dir_ in os.listdir(self.models_dir) if os.path.isdir(os.path.join(self.models_dir, dir_))]`

should do the trick.

Comment (QuickLearner171998) [reply main.py:215]: Yeah... I also noticed it now only...thanks

**Comment by virajmavani** [comment]
These are some great changes! Thank you for contributing to the project!

Once the changes have been made, I will go ahead and merge.

Later, I am planning to refactor the whole project as it is abominable. I don't know what I was thinking when I wrote it 2 years back.

If interested, we can make a Slack group for developers and chat over it so that anyone else can also join.

**Comment by QuickLearner171998** [comment]
Yes, that would be great!!

**Comment by virajmavani** [comment]
Looks good enough. I will merge. Good job!

**Pull request merged**